nix/corsixth.nix

```diff
@@ -0,0 +1,89 @@
+{ baseName ? "corsix-th", stdenv, cmake, lib, SDL2, lua5_3, makeWrapper
+, SDL2_mixer ? null, soundfont-fluid ? null, fluidsynth ? null, ffmpeg ? null, freetype ? null, doxygen ? null, catch2 ? null
+, audioSupport ? true, movieSupport ? true, freetypeSupport ? true, buildDocs ? false, enableUnitTests ? false
+}:
+
+assert audioSupport -> SDL2_mixer != null && soundfont-fluid != null && fluidsynth != null;
+assert movieSupport -> ffmpeg != null;
+assert freetypeSupport -> freetype != null;
+assert buildDocs -> doxygen != null;
+assert enableUnitTests -> catch2 != null;
+
+with lib;
+
+let
+    version = "0.64";
+
+    lua = {
+        env = lua5_3.withPackages(ps: with ps; [ lpeg luafilesystem luasocket ]);
```

Comment:
luasocket is currently only used for the update check. If you want to depend on the package manager you can leave it out.

Reply:
I was considering to remove it, just was not sure if it could cause issues down the line. All I know is we need to update through the package manager.